lib/improver/utilities/spatial.py

```diff
@@ -49,54 +49,77 @@
 MAX_DISTANCE_IN_GRID_CELLS = 500
 
 
-def check_if_grid_is_equal_area(cube):
+def check_if_grid_is_equal_area(cube, require_equal_xy_spacing=True):
     """Identify whether the grid is an equal area grid.
     If not, raise an error.
     Args:
         cube (iris.cube.Cube):
-            Cube with coordinates that will be cMAhecked.
+            Cube with coordinates that will be checked.
+        require_equal_spacing (bool):
+            Flag to require the grid is equally spaced in the two spatial
+            dimensions (not strictly required for equal-area criterion)
+
+    Raises:
+        ValueError: If coordinate points are not equally spaced along either
+            axis (from calculate_grid_spacing)
+        ValueError: If point spacing is not equal for the two spatial axes
+    """
+    xdiff = calculate_grid_spacing(cube, axis='x')
+    ydiff = calculate_grid_spacing(cube, axis='y')
+    if require_equal_xy_spacing and not np.isclose(xdiff, ydiff):
+        raise ValueError(
+            "Grid does not have equal spacing in x and y dimensions")
```

Comment:
This function is much tidier. The keyword argument is redundant as far as I can see; if you call it with anything other than True the function does nothing, so why would we call it? This is the reason there are no tests for the False option I presume.

We definitely want to separate equal-areas and equal-grid-spacing into two functions that do what they say on the tin.

Reply:
The `calculate_grid_spacing` function raises a `ValueError` with a useful message if either coordinate has non-equally-spaced grid points.  Since it's a useful error message I've not caught it here but just allow it to be raised.  So the function does genuinely check for an equal area grid, more literally if you set this to `False`.  There is a unit test for this (in the case where this statement would be reached); see the two tests at the end of the test class.